File: Offer/16/Solution.go
package Solution

type ListNode struct {
	Val  int
	Next *ListNode
}

/**
 *
 * @param pHead1 ListNode类
 * @param pHead2 ListNode类
 * @return ListNode类
 */
func Merge(pHead1 *ListNode, pHead2 *ListNode) *ListNode {
	if pHead1 == nil {
		return pHead2
	}
	if pHead2 == nil {
		return pHead1
	}
	dummy := &ListNode{
		0,
		nil,
	}
	curr := dummy
	for {
		if pHead1 == nil || pHead2 == nil {
			break
		}
		if pHead1.Val > pHead2.Val {
			curr.Next = pHead2
			pHead2 = pHead2.Next
		} else {
			curr.Next = pHead1
			pHead1 = pHead1.Next
		}
		curr = curr.Next
	}
	if pHead1 != nil {
		curr.Next = pHead1
	}
	if pHead2 != nil {
		curr.Next = pHead2
	}
	return dummy.Next
}

File: Offer/16/Solution.java

public class ListNode {
    int val;
    ListNode next = null;

    ListNode(int val) {
        this.val = val;
    }
}
public class Solution {
    public ListNode Merge(ListNode list1,ListNode list2) {
        if(list1 == null || list2 == null) return (list1 == null) ? list2 : list1;
        ListNode dummy = new ListNode(0);
        ListNode curr = dummy;
        while(list1 != null && list2 != null){
            if(list1.val > list2.val) {
                curr.next = list2;
                list2 = list2.next;
            }else{
                curr.next = list1;
                list1 = list1.next;
            }
            curr = curr.next;
        }
        if(list1 != null) curr.next = list1;
        if(list2 != null) curr.next = list2;
        return dummy.next;
    }
}